pdns/recpacketcache.cc

```diff
@@ -174,6 +178,7 @@ void RecursorPacketCache::insertResponsePacket(unsigned int tag, uint32_t qhash,
     e.d_ttd = now+ttl;
     e.d_creation = now;
     e.d_tag = tag;
+    e.d_vstate = valState;
```

Comment:
We should likely also update the validation state when we replace an existing entry, just above.